third_party/musl/crt/patch.diff

```diff
@@ -243,15 +271,40 @@ index afec985a..0ac9f0c4 100644
  static uint32_t sysv_hash(const char *s0)
  {
  	const unsigned char *s = (void *)s0;
-@@ -895,6 +910,7 @@ static int fixup_rpath(struct dso *p, char *buf, size_t buf_size)
+@@ -658,6 +678,24 @@ static void *map_library(int fd, struct dso *dso)
+ 					ph->p_memsz < DEFAULT_STACK_MAX ?
+ 					ph->p_memsz : DEFAULT_STACK_MAX;
+ 			}
++			if(myst_exec_stack_support()) {
++				int prot_flags = 0;
++				if (ph->p_flags & PF_X)
++					prot_flags |= PROT_EXEC;
++				
++				if (ph->p_flags & PF_W)
++					prot_flags |= PROT_WRITE;
++				
++				if (ph->p_flags & PF_R)
++					prot_flags |= PROT_READ;
++				
++				void *stack_p = 0;
++				size_t *stack_size = 0;
++				if (!syscall(SYS_get_process_thread_stack, &stack_p, &stack_size))
```

Comment:
I am unsure if SYS_get_process_thread_stack is going to work in all scenarios.
This was written for use in fork(). In fork the call can be called from any thread and it first checks the pthread structure to see if it has a stack from present there. Then if that does not exist it falls back to this syscall. 
With the first process to execute the kernel allocates the stack. This stack is returned from the syscall.
I cannot remember the situation for all other situations though regarding the process thread of child processes.
You will need to check if a fork/exec or spawned child process can also access the stack with this syscall, or if you need to retrieve it from the pthread structure